### PR TITLE
Add check for docker engine liveness in preflight

### PR DESCRIPTION
**Pull request by odinuge**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

If docker engine/server is not running, the preflight check will now report:

```
system validation failed: [unsupported docker version: ]
```

because the output of `docker info --format "{{json .}}"` is:

```json
{
  "ServerErrors": [
    "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
  ],
  "ClientInfo": {
    "Debug": false,
    "Plugins": [],
    "Warnings": null
  }
}
```

With this PR the check will report:

```
system validation failed: [error when fetching docker info: "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"]
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

/assign random-liu

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/util/system/docker_validator.go [hunk 1]
```diff
@@ -81,6 +84,10 @@ func (d *DockerValidator) unmarshalDockerInfo(b []byte, info *dockerInfo) error
 
 func (d *DockerValidator) validateDockerInfo(spec *DockerSpec, info dockerInfo) ([]error, []error) {
 	// Validate docker version.
+	if info.ServerErrors != nil {
+		return nil, []error{errors.Errorf(`error verifying docker info: "%s"`, strings.Join(info.ServerErrors, `", "`))}
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/docker_validator.go]: `error verifying docker info: "%s"` can be changed to "error verifying docker info: %q"
notice, not using `

**Comment by neolit123** [comment]
/hold
@odinuge 
we just moved the validators to the kubernetes/system-validators repository, so you should send the PR there. we are about to remove them from kubernetes/kubernetes.

so please send the PR for that repository and let's close this PR for kubernetes/kubernetes.

File: cmd/kubeadm/app/util/system/docker_validator.go (hunk 1 above)
Comment (odinuge) [reply cmd/kubeadm/app/util/system/docker_validator.go]: Thanks, totally forgot about `%q`

**Comment by odinuge** [comment]
Aha, that make sense!

Filed https://github.com/kubernetes/system-validators/pull/3/files

/close

**Pull request closed without merging**